tests/scalapack/scalapack_01.cc

```diff
@@ -0,0 +1,132 @@
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#include "../tests.h"
+
+// test copying FullMatrix into distributed ScaLAPACK matrix and back.
+
+#include <deal.II/base/logstream.h>
+#include <deal.II/base/utilities.h>
+#include <deal.II/base/conditional_ostream.h>
+#include <deal.II/base/timer.h>
+#include <deal.II/base/multithread_info.h>
+
+#include <boost/random/mersenne_twister.hpp>
+#include <boost/random/uniform_01.hpp>
+
+#include <deal.II/lac/vector.h>
+
+#include <deal.II/lac/scalapack.h>
+
+#include <fstream>
+#include <iostream>
+
+template <typename NumberType>
+void test(const unsigned int size, const unsigned int block_size)
+{
+  MPI_Comm mpi_communicator(MPI_COMM_WORLD);
+  const unsigned int n_mpi_processes(Utilities::MPI::n_mpi_processes(mpi_communicator));
+  const unsigned int this_mpi_process(Utilities::MPI::this_mpi_process(mpi_communicator));
+
+  ConditionalOStream pcout (std::cout, (this_mpi_process ==0));
+
+  // test multiplication with random vectors
+  boost::random::mt19937 gen;
+  boost::random::uniform_01<> dist;
+
+  // Create SPD matrices of requested size:
+  FullMatrix<NumberType> full_in(size), full_out(size), diff(size);
+
+  std::pair<int,int> sizes = std::make_pair(size,size), block_sizes = std::make_pair(block_size,block_size);
+  std::shared_ptr<ProcessGrid> grid = std::make_shared<ProcessGrid>(mpi_communicator,sizes,block_sizes);
+
+  ScaLAPACKMatrix<NumberType> scalapack_matrix (sizes.first, grid, block_sizes.first);
+  /*ScaLAPACKMatrix<NumberType> scalapack_matrix(size,
+                                               mpi_communicator,
+                                               block_size);*/
+
+  pcout << size << " " << block_size << " " << grid->get_process_grid_rows() << " " << grid->get_process_grid_columns() << std::endl;
+
+  {
+    unsigned int index = 0;
+    for (unsigned int i = 0; i < size; ++i)
+      for (unsigned int j = 0; j < size; ++j)
+        full_in(i,j) = index++;
+  }
+
+  scalapack_matrix = full_in;
+  scalapack_matrix.copy_to(full_out);
+
+  diff = 0;
+  diff.add( 1., full_in);
+  diff.add(-1., full_out);
+
+  const double error = diff.frobenius_norm();
+
+  if (error > 1e-10 && this_mpi_process == 0)
+    {
+      std::cout << "Error!" << std::endl
+                << "expected to have:" << std::endl;
+      full_in.print_formatted (std::cout);
+      std::cout << "but got:" << std::endl;
+      full_out.print_formatted(std::cout);
+      AssertThrow (false, dealii::ExcInternalError());
+    }
+  else
+    pcout << "Ok" << std::endl;
+}
+
+
+
+int main (int argc,char **argv)
+{
+
+  try
+    {
+      Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv, numbers::invalid_unsigned_int);
+
```

Comment:
Our tests typically use ``MPILogInitAll`` class and ``deallog`` for output. Can you please change to that?

Reply:
that's not needed here, we don't have output from all MPI cores, only master writes "Ok" if no errors are there.